api/docs/release.dox

```diff
@@ -126,7 +126,8 @@ clients.
 
 The changes between version \DR_VERSION and 11.3.0 include the following compatibility
 changes:
- - No compatibility changes yet.
+ - The size of #dr_mcontext_t on 32-bit Arm has been increased by 4 to
+   make it 8-byte aligned.
```

Comment:
This makes it sound like we appended padding to the end where a recompile is enough; but really we inserted padding in between fields, so any code such as generated machine instructions for instrumentation has to be updated as the field offsets have changed.  I think this needs to explicitly say that some field offsets have changed.

Reply:
I've improved the change log ... and I've just realised that I hadn't actually made `priv_mcontext_t` 8-byte aligned! I'd changed its size to a multiple of 8, and in the places where it is allocated in assembly on the stack it is 8-byte aligned because SP is 8-byte aligned, and it would be automatically 8-byte aligned if someone were to use `malloc`, but a `priv_mcontext_t` local variable might not be 8-byte aligned as it was. So I've added a `uint64 _unused` in an anonymous union in the struct definition. Do you think that's the thing to do and the right way to do it?

Reply:
> So I've added a uint64 _unused in an anonymous union in the struct definition. Do you think that's the thing to do and the right way to do it?

Use `ALIGN_VAR(8)`.

Reply:
I don't think you marked `priv_mcontext_t` as `ALIGN_VAR`, so it won't be aligned, right?

Reply:
Yes, I may have been a bit hasty in merging that, and the change log entry is unclear: I think I honestly meant "it" to refer to the "size" when I wrote that but readers will, of course, understand "it" to refer to the struct!

Enforcing alignment seems to cause a problem with cross-compilation that I'm not yet set up to reproduce on a local machine: #7196.

We must either debug that cross-compilation problem and enforce alignment, or correct that log message to make it clear that all we have done is make the size a multiple of 8 in order to facilitate alignment.

(The cross-compilation problem seems rather interesting. If there are lingering alignment issues in the code I wonder if #7161 will turn out to be alignment-related.)